### PR TITLE
Removed swig from dependencies

### LINKED ISSUE #219: Fix installation process
This issue is to collect some fix need to make in the installation script.

The target is to make the installation work on Raspbian Stretch (9) and Ubuntu 18.04, and make the process simpler.

Some core idea:
- Try to install the dependency software from the OS repository (via `apt`) and from Python [PyPI](https://pypi.org/) (via `pip3`) first. Only install from source (`make`) if the previous method fails.

ISSUES To be solved:
- [ ] SWIG should be installed from Debian repository.
- [x] libatlas-dev doesn't exist. Install libatlas-base-dev instead.
- [x] Tizonia default installation script only installs Python 2 packages (from https://dl.bintray.com/tizonia/raspbian), not Py3. Doesn't have a fix yet!
- [x] The final step in install.sh doesn't prompt the user to use python3 config_generator.py however, the new format is 'python3 config_generator.py stt tts hotword wake'
- [x] The 99-com.rules file is of Raspbian. We must not write our rule to it. Write to new file, so that when we uninstall the app, we just delete our file.
- [x] Some Python packages, like snowboy, has to be installed from Git source, just because the version published to PyPI is too old.
- [x] Try to install the dependency software from the OS repository (via `apt`) and from Python [PyPI](https://pypi.org/) (via `pip3`) first. Only install from source (`make`) if the previous method fails.

### PR DESCRIPTION
**Pull request by stealthanthrax**
Completly Fixes #219 

Removed Swig's source installation and using OS repo's to install it

### CODE REVIEW

File: auto_boot/auto_boot.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+SCRIPT_PATH=$(realpath $0)
+DIR_PATH=$(dirname $SCRIPT_PATH)
+
+mv $DIR_PATH/media.service /lib/systemd/system/media.service
+mv $DIR_PATH/update.service /lib/systemd/system/update.service
+
+sudo systemctl daemon-reload
+sudo systemctl enable /lib/systemd/system/media.service
+sudo systemctl enable /lib/systemd/system/update.service
```
Comment (hongquan) [inline auto_boot/auto_boot.sh]: Again, no new line at the end of file???

Comment (hongquan) [reply auto_boot/auto_boot.sh]: No need to specify full path to `systemctl enable`.

**Comment by hongquan** [comment]
@stealthanthrax Why are there service files of different issue in this PR?

Why do you name the service so vague "media.service", "update.service"? Didn't I have requirement about service name already?

File: install.sh [hunk 2]
```diff
@@ -77,15 +67,14 @@ function install_dependencies()
     then
         if ! [ -x "$(command -v swig)" ]
         then
-            install_swig_from_sources
+            echo "swig not installed"
         else
             installed_version=$(swig -version | perl -nae 'print "$F[2]\n" if /SWIG Version/i;')
-            minimum_version="3.0.10"
+            minimum_version="3.0.8"
             result=$(vercomp ${installed_version} ${minimum_version})
```
Comment (hongquan) [inline install.sh]: Any reason why there is comparing of Swig here?

**Comment by stealthanthrax** [comment]
@hongquan , I have updated the files

File: install.sh [hunk 3]
```diff
@@ -77,15 +67,14 @@ function install_dependencies()
     then
         if ! [ -x "$(command -v swig)" ]
         then
-            install_swig_from_sources
+            echo "swig not installed"
         else
             installed_version=$(swig -version | perl -nae 'print "$F[2]\n" if /SWIG Version/i;')
             minimum_version="3.0.10"
             result=$(vercomp ${installed_version} ${minimum_version})
```
Comment (hongquan) [inline install.sh]: Why do you have to compare the version here?
SWIG is installed from OS repo, it cannot have version lower than the one in repo. And we don't support Debian older than 9, so the Swig won't be older than 3.0.12.
No need to compare here.

**Comment by stealthanthrax** [comment]
@hongquan , i have updated it

**Comment by woshikie** [comment]
@stealthanthrax 
I don't see it

https://github.com/stealthanthrax/susi_linux/blob/0db73f7d6038b6d8a0827dc95b384eeaa487afa8/install.sh#L84

The function name is still the same here,

https://github.com/stealthanthrax/susi_linux/blob/0db73f7d6038b6d8a0827dc95b384eeaa487afa8/install.sh#L104

we're still not sure to remove this or not

**Comment by cweitat** [comment]
@stealthanthrax pls fix the conflict

**Comment by stealthanthrax** [comment]
@cweitat , updated

**Comment by stealthanthrax** [comment]
@hongquan , can it be merged now ??

**Pull request merged**